File: examples/languages/sample.ts
// TypeScript sample demonstrating syntax highlighting
interface Color {
  r: number;
  g: number;
  b: number;
}

class Palette {
  private colors: Map<string, Color>;

  constructor() {
    this.colors = new Map();
  }

  add(name: string, color: Color): void {
    this.colors.set(name, color);
  }

  toHex(color: Color): string {
    const r = color.r.toString(16).padStart(2, '0');
    const g = color.g.toString(16).padStart(2, '0');
    const b = color.b.toString(16).padStart(2, '0');
    return `#${r}${g}${b}`;
  }

  display(): void {
    this.colors.forEach((color, name) => {
      console.log(`${name}: ${this.toHex(color)}`);
    });
  }
}

// Main execution
const palette = new Palette();
palette.add("red", { r: 255, g: 0, b: 0 });
palette.add("green", { r: 0, g: 255, b: 0 });
palette.add("blue", { r: 0, g: 0, b: 255 });
palette.display();

File: examples/languages/sample.rs
// Rust sample demonstrating syntax highlighting
use std::collections::HashMap;

/// Calculate the factorial of a number
fn factorial(n: u64) -> u64 {
    match n {
        0 | 1 => 1,
        _ => n * factorial(n - 1),
    }
}

#[derive(Debug, Clone)]
struct Color {
    r: u8,
    g: u8,
    b: u8,
}

impl Color {
    fn new(r: u8, g: u8, b: u8) -> Self {
        Self { r, g, b }
    }

    fn to_hex(&self) -> String {
        format!("#{:02x}{:02x}{:02x}", self.r, self.g, self.b)
    }
}

fn main() {
    let mut colors = HashMap::new();
    colors.insert("red", Color::new(255, 0, 0));
    colors.insert("green", Color::new(0, 255, 0));
    colors.insert("blue", Color::new(0, 0, 255));

    for (name, color) in &colors {
        println!("{}: {}", name, color.to_hex());
    }

    let result = factorial(5);
    println!("5! = {}", result);
}
